cqc/pythonLib.py

```diff
@@ -1737,13 +2110,28 @@ def _single_gate_rotation(self, command, step, notify, block):
         self.check_active()
 
         if self._cqc.pend_messages:
+
+            # If we are inside a TP_MIX, then insert the CQC Type header before the command header
```

Comment:
It would be good to put these things in functions since they can then be reused in `_two_qubit_gate`, `measure` `reset` etc

Reply:
Do you mean only these two lines? 

```
if self._cqc._inside_cqc_mix:
    self._cqc._pend_type_header(CQCType.COMMAND, CQCCmdHeader.HDR_LENGTH + CQCAssignHeader.HDR_LENGTH)
```

Reply:
There is also this:
```python
            # Build command header and rotation sub header
            command_header = CQCCmdHeader()
            command_header.setVals(self._qID, command, notify, block)
```
and this
```python
            # Pend headers
            self._cqc._pend_header(command_header)
            self._cqc._pend_header(rot_sub_header)
```
copied code

Reply:
Done